airbyte-webapp/src/core/resources/BaseResource.ts

```diff
@@ -67,9 +76,11 @@ export default abstract class BaseResource extends Resource {
     return config.apiUrl;
   }
 
-  static listShape<T extends typeof Resource>(this: T) {
+  static listShape<T extends typeof Resource>(
+    this: T
+  ): ReadShape<SchemaList<AbstractInstanceType<T>>> {
     return {
-      ...super.listShape(),
+      ...(super.listShape() as any),
```

Comment:
Not sure that `as any` is a good option, but I am totally fine using it this way for now.

Reply:
It's an unfortunate limitation due to `super` generics bug. I've sadly gone deep into this trail. Generics in this case are kind of undocumented I think because the TypeScript team hasn't fleshed out all the issues. I'll try to get the TypeScript issue linked to in an updated documentation around this.